src/Compilers/CSharp/Portable/Parser/LanguageParser.cs

```diff
@@ -878,7 +878,30 @@ private static bool IsGlobalAttributeTarget(SyntaxToken token)
 
         private bool IsPossibleAttributeDeclaration()
         {
-            return this.CurrentToken.Kind == SyntaxKind.OpenBracketToken;
+            // Have to at least start with `[` to be an attribute
+            if (this.CurrentToken.Kind != SyntaxKind.OpenBracketToken)
+                return false;
+
+            using var _ = this.GetDisposableResetPoint(resetOnDispose: true);
+
+            // Eat the `[`
+            EatToken();
+
+            // `[ id` is always an attribute.
+            if (this.IsTrueIdentifier())
+                return true;
+
+            // `[ word: ...` could be an attribute.
+            if (IsAttributeTarget())
+                return true;
+
+            // If we see `[lit` (like `[0`) then this is def not an attribute, and should be parsed as a collection
+            // expr.  Note: this heuristic can be added to in the future.
+            var isLiteral = SyntaxFacts.GetLiteralExpression(this.CurrentToken.Kind) != SyntaxKind.None;
```

Comment:
>GetLiteralExpression [](http://example.com/codeflow?start=40&length=20)

Consider using ```IsLiteralExpression``` helper instead. It matches our intent exactly.

Reply:
Ah.  I missed that that exists.  THanks.